### ISSUE TITLE
When using RestyJsonTypeIdResolver calls return nulls instead of constructed objects

### ISSUE THREAD

**Issue by Oberonc** [resty-gwt/resty-gwt#432]
Hi,

When I use RestyJsonTypeIdResolver to resolve types dynamically REST calls return nulls instead of objects where it should.

I know I'm putting the right classes into the map returned by "getIdClassMap()" because when I omit one I get the following exception in the browser console:
Exception: com.github.nmorel.gwtjackson.client.exception.JsonDeserializationException: Could not find the type associated to [my class].

So in other words the code resolves my classes, but doesnt construct them and returns nulls.

Also the server side works correctly (I can see that the calls return correct data in the browser console)
So server side TypeIdResolver is correct (and is returned inside my class that extends RestyJsonTypeIdResolver).

I'm using RestyGWT 2.2.7
GWT 2.9
jaxb-api 2.2.12
javax.ws.rs-api 2.0.1

Thanks
